source/Mlos.Python/mlos/Optimizers/OptimizerBase.py

```diff
@@ -107,10 +113,30 @@ def optimum(self, optimum_definition: OptimumDefinition = OptimumDefinition.BEST
             if context is None:
                 raise ValueError(f"{optimum_definition} requires context to be not None.")
             return self._optimum_within_context(context=context)
+        if optimum_definition == OptimumDefinition.BEST_SERIES_ERROR_FUNCTION_MINIMIZATION:
+            if context is None:
+                raise ValueError(f"{optimum_definition} requires context to be not None.")
+            return self._optimum_with_series_difference(context=context)
         return self._prediction_based_optimum(parameters_df=parameters_df, optimum_definition=optimum_definition, alpha=alpha)
 
+    @trace()
+    def _optimum_with_series_difference(self, context: pd.DataFrame):
+        predicted_value_utility_function = SeriesPredictedValueUtilityFunction(
+            surrogate_model=self.surrogate_model,
+            objective=self.optimization_problem.objectives[0],
+            logger=self.logger
+        )
+        utility_optimizer = UtilityFunctionOptimizerFactory.create_utility_function_optimizer(
+            utility_function=predicted_value_utility_function,
+            optimizer_type_name=RandomSearchOptimizer.__name__,
+            optimizer_config=random_search_optimizer_config_store.default,
+            optimization_problem=self.optimization_problem
+        )
+        return utility_optimizer.suggest(context_values_dataframe=context)
+
     @trace()
     def _optimum_within_context(self, context: pd.DataFrame):
+        # TODO ZACK: Discuss the minimize=... line. Why only objectives[0]?

```

Comment:
Since nobody ever added support for context to work in multi-objective settings.

Reply:
When I saw this, I became very sad because I spent a great deal of time with context-optimization problems that have mutiple objectives... I got terrible performance.